### PR TITLE
Make AsyncHttpClient.apply return an effect

### PR DESCRIPTION
**Pull request by rossabaker**
Includes merge of #1776.

### CODE REVIEW

File: async-http-client/src/test/scala/org/http4s/client/asynchttpclient/AsyncHttpClientSpec.scala [hunk 1]
```diff
@@ -7,4 +7,4 @@ import cats.effect.{Effect, IO}
 class AsyncHttpClientSpec
     extends ClientRouteTestBattery(
       "AsyncHttpClient",
-      AsyncHttpClient()(Effect[IO], Http4sSpec.TestExecutionContext))
+      AsyncHttpClient()(Effect[IO], Http4sSpec.TestExecutionContext).unsafeRunSync)
```
Comment (rossabaker) [inline async-http-client/src/test/scala/org/http4s/client/asynchttpclient/AsyncHttpClientSpec.scala]: Ick.  It's already this way in blaze.

**Comment by jmcardon** [comment]
Just for record-keeping, [discussion on gitter](https://gitter.im/http4s/http4s?at=5ace49e4080a3850531ffdad) re: wanting to backport

**Comment by jmcardon** [comment]
Should we close this and merge up instead?

**Comment by jmcardon** [comment]
I approved this, since it's definitely better than the status quo, but I wonder whether we're gonna consider the possible `allocate[F]` and `release[F]` or `close[F]` convention?

**Comment by rossabaker** [comment]
Yes, I think we should.  Otherwise, we're just adding a method we'd like to deprecate in blaze.

**Comment by rossabaker** [comment]
Bitrotten.

**Pull request closed without merging**